tests/waku_archive/test_retention_policy.nim

```diff
@@ -53,6 +54,45 @@ suite "Waku Archive - Retention policy":
 
     ## Cleanup
     (waitFor driver.close()).expect("driver to close")
+  
+  test "size retention policy - windowed message deletion":
+    ## Given
+    let
+      # in megabytes
+      sizeLimit:float = 0.05
+      excess = 123
+
+    let driver = newTestArchiveDriver()
+
+    let retentionPolicy: RetentionPolicy = SizeRetentionPolicy.init(size=sizeLimit)
+
+    ## When
+
+    var putFutures = newSeq[Future[ArchiveDriverResult[void]]]()
+    var retentionFutures = newSeq[Future[ArchiveDriverResult[void]]]()
+
+    for i in 1..excess:
+        let msg = fakeWakuMessage(payload= @[byte i], contentTopic=DefaultContentTopic, ts=Timestamp(i))
+        putFutures.add(driver.put(DefaultPubsubTopic, msg, computeDigest(msg), msg.timestamp))
+        retentionFutures.add(retentionPolicy.execute(driver))
+
+    # waitFor is used to synchronously wait for the futures to complete.
+    discard waitFor allFinished(putFutures & retentionFutures)
+            
+    ## Then
+    # calculate the current database size
+    let pageSize = (waitFor driver.getPagesSize()).tryGet()
```

Comment:
Good point Ivan,
Changes made to compare the current db size with the size limit, retention policy enabled only in case of overflow.
After the retention policy is executed, again comparison is made between the current db size and the size limit.